scripts/sequential_learning.py

```diff
@@ -120,6 +140,20 @@ def choose_next_candidate(
         parent_idx = np.arange(unct.shape[0])[~train_mask][next_idx]
         # get the actual value of the MU score for record keeping
         max_score = np.max(unct[~train_mask])
+    elif acquisition_function.lower() == "space-filling":
+        feature_columns = [c for c in df.columns if c not in non_feature_columns]
+        X = np.array(df[feature_columns].values)
+        # scale the data (for using distance-based metrics)
+        scaler = StandardScaler()
+        X = scaler.fit_transform(X)
+        # calculate all pairwise distances between examples (feature vectors)
+        D = euclidean_distances(X, X)
+        # print(f"D.shape: {D.shape}")
+        # choose the candidate in the design space that is farthest away from the
+        # training set
```

Comment:
can we rephrase slightly to try and make the logic of the below
lines a little more obvious? maybe something like below?
```suggestion
        # choose the unlabelled candidate in the design space whose 
        # closest neighbor in the training set is the farthest away
```